From f48883bca782b409f4b9e1b693df5a10f95b2ddb Mon Sep 17 00:00:00 2001
From: Gustavo Zacarias <gustavo@zacarias.com.ar>
Date: Mon, 3 Mar 2014 10:40:08 -0300
Subject: ui: switch to new-style readline typedef

The CPPFunction typedef (among others) have been deprecated in favour of
specific prototyped typedefs since readline 4.2 (circa 2001).
It's been working since because compatibility typedefs have been in
place until they where removed in the recent readline 6.3 release.
Switch to the new style to avoid build breakage.

Signed-off-by: Gustavo Zacarias <gustavo@zacarias.com.ar>
Origin: upstream, http://git.savannah.gnu.org/cgit/parted.git/commit/?id=cc382c3753e1b430c00df46f0937b74db36ccb8a
Last-Update: 2014-03-28

Patch-Name: new-style-readline-typedef.patch
---
 parted/ui.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/parted/ui.c b/parted/ui.c
index d00c65a..a4f7222 100644
--- a/parted/ui.c
+++ b/parted/ui.c
@@ -1474,7 +1474,7 @@ init_readline (void)
 #ifdef HAVE_LIBREADLINE
   if (!opt_script_mode) {
     rl_initialize ();
-    rl_attempted_completion_function = (CPPFunction*) complete_function;
+    rl_attempted_completion_function = (rl_completion_func_t *) complete_function;
     readline_state.in_readline = 0;
   }
 #endif
